sim.py

```diff
@@ -5,94 +5,93 @@ class YieldSimulator(object):
             'base_asset_price',
             'vault_apy',
-            'pool_age',
             'x_reserves',
             'y_reserves',
             'total_supply',
             'token_in',
             'token_out',
             'trade_direction',
             'trade_amount',
+            'trade_amount_usd',
             'price_per_share', # c in YieldSpace with Yield Bearing Vaults
             'init_price_per_share', # u in YieldSpace with Yield Bearing Vaults
             'out_without_fee_slippage',
             'out_with_fee',
             'out_without_fee',
             'fee',
+            'slippage',
             'days_until_maturity',
             'num_trading_days',
+            'random_seed',
             'day',
+            'trade_number',
             'spot_price',
-            'num_orders'
+            'num_orders',
+            'step_size',
         ]
         self.analysis_dict = {key:[] for key in analysis_keys}
         self.random_variables_set = False
-        self.set_random_variables()
 
     def set_random_variables(self):
+        self.rng = np.random.default_rng(self.random_seed)
         self.target_liquidity = self.rng.uniform(self.min_target_liquidity, self.max_target_liquidity)
         self.target_daily_volume = self.rng.uniform(self.min_target_volume, self.max_target_volume)
-        self.start_apy = self.rng.uniform(self.min_apy, self.max_apy) # starting fixed apr
+        self.init_pool_apy = self.rng.uniform(self.min_pool_apy, self.max_pool_apy) # starting fixed apr
         self.fee_percent = self.rng.uniform(self.min_fee, self.max_fee)
         # Determine real-world parameters for estimating u and c (vault and pool details)
-        # TODO: Should vault_age be used to set u instead of pool_age?
         self.init_vault_age = self.rng.uniform(self.min_vault_age, self.max_vault_age) # in years
         self.vault_apy = self.rng.uniform(self.min_vault_apy, self.max_vault_apy, size=self.num_trading_days) / 100 # as a decimal
-        # TODO: pool_age is probably not correctly named, and could just be a function of days_until_maturity
-        self.pool_age = self.rng.uniform(min(self.init_vault_age, self.min_pool_age), self.max_pool_age) # in years
         self.random_variables_set = True
 
     def print_random_variables(self):
         print('Simulation random variables:\n'
             + f'target_liquidity: {self.target_liquidity}\n'
             + f'target_daily_volume: {self.target_daily_volume}\n'
-            + f'start_apy: {self.start_apy}\n'
+            + f'init_pool_apy: {self.init_pool_apy}\n'
             + f'fee_percent: {self.fee_percent}\n'
             + f'init_vault_age: {self.init_vault_age}\n'
-            + f'init_vault_apy: {self.vault_apy[0]}\n'
-            + f'pool_age: {self.pool_age}\n'
+            + f'init_vault_apy: {self.vault_apy[0]}\n' # first element in vault_apy array
         )
 
     def run_simulation(self, override_dict=None):
+        self.trade_number = 0
```

Comment:
@wakamex can you rename this to run_trade_number

Reply:
done